src/nvim/tui/tui.c

```diff
@@ -0,0 +1,718 @@
+#include <assert.h>
+#include <stdbool.h>
+#include <stdio.h>
+
+#include <unistd.h>
+#include <sys/ioctl.h>
+
+#include <uv.h>
+#include <termkey.h>
+#include <unibilium.h>
+
+#include "nvim/lib/kvec.h"
+
+#include "nvim/vim.h"
+#include "nvim/ui.h"
+#include "nvim/memory.h"
+#include "nvim/api/vim.h"
+#include "nvim/api/private/helpers.h"
+#include "nvim/os/event.h"
+#include "nvim/os/input.h"
+#include "nvim/tui/tui.h"
+
+typedef struct {
+  int top, bot, left, right;
+} Rect;
+
+typedef struct {
+  char data[8];
```

Comment:
Why is this array `8` chars big? Longest UTF-8 + NUL would be at most `7` chars.